test/caPutJsonLogTest.cpp

```diff
@@ -471,6 +489,12 @@ void commonTests(JsonParser &jsonParser, const char* pvname, const char * testPr
 
     testOk(!jsonParser.pv.compare(pvname),
             "%s - %s", testPrefix, "PV name check");
+
+    //Test Metadata
+    testOk(logger->metadataCount() == jsonParser.metadata.size(),
+            "%s - %s (%u vs %lu)", testPrefix, "Metadata elements count check", logger->metadataCount(), jsonParser.metadata.size());
+    testOk(jsonParser.metadata_compare(logger->getMetadata()),
+            "%s - %s", testPrefix, "Metadata keys and values check");
```

Comment:
I understand here you are checking if the metadata added to the logger is the same as you get on the end message right?
A test is missing to check if the metadata you've added is in the end message. What you are doing here is test if what is on the metadata is also on the end json message. But you never test if the intended metadata is on the end json message.

Reply:
I have a map in JsonParser which is constructed with the metadata  that is parsed from the message(anything that doesn't have any of the keywords which are already in use like "pv", "host", etc and it's in the logger object). So what I do in the metadata_compare is just to check if both maps match(the one from the logger and the one rebuild by the json parser).

   bool metadata_compare (std::map<std::string, std::string> ext_metadata) {
       return this->metadata.size() == ext_metadata.size()
       && std::equal(this->metadata.begin(), this->metadata.end(), ext_metadata.begin());
   }